src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/GetAzureContainerRegistry.cs

```diff
@@ -0,0 +1,84 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsCommon.Get, ContainerRegistryNoun), OutputType(typeof(PSContainerRegistry))]
+    public class GetAzureContainerRegistry : ContainerRegistryCmdletBase
+    {
+        internal const string ResourceGroupParameterSet = "ResourceGroupParameterSet";
+        internal const string RegistryNameParameterSet = "RegistryNameParameterSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = false,
+            ParameterSetName = ResourceGroupParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = RegistryNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = RegistryNameParameterSet,
+            HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!string.IsNullOrEmpty(ResourceGroupName) && !string.IsNullOrEmpty(Name))
+            {
+                var registry = RegistryClient.GetRegistry(ResourceGroupName, Name);
+                WriteObject(new PSContainerRegistry(registry));
+            }
+            else
+            {
+                var registries = RegistryClient.ListRegistries(ResourceGroupName);
+
+                List<PSContainerRegistry> list = new List<PSContainerRegistry>();
+                foreach (Registry registry in registries)
+                {
+                    list.Add(new PSContainerRegistry(registry));
+                }
+                WriteObject(list, true);
+
+                while (!string.IsNullOrEmpty(registries.NextPageLink))
+                {
+                    registries = RegistryClient.ListRegistriesUsingNextLink(ResourceGroupName, registries.NextPageLink);
+
+                    list = new List<PSContainerRegistry>();
+                    foreach (Registry registry in registries)
+                    {
+                        list.Add(new PSContainerRegistry(registry));
+                    }
+                    WriteObject(list, true);
```

Comment:
@djyou I'm not 100% sure what happens when you write an enumerated collection to the pipeline a page at a time, but I would recommend doing one of the following:

1) Populate the `registries` list with every page of the results, and then write to the pipeline _once_ after the while loop has terminated.

2) For the individual objects to the pipeline rather than a page at a time. For example, the while loop would instead look like the following (as well as the section before the while loop):

```cs
while (!string.IsNullOrEmpty(registries.NextPageLink))
{
    registries = RegistryClient.ListRegistriesUsingNextLink(ResourceGroupName, registries.NextPageLink);
    foreach (Registry registry in registries)
    {
        WriteObject(new PSContainerRegistry(registry);
    }
}
```

Both of these scenarios try to avoid the possibility of writing groups of objects to the pipeline and causing unusual behavior. It would be better to write each object as you get them, or write all objects at once.